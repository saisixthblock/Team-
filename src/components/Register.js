import React, { Component } from "react";
import {Card, CardBody, CardHeader,Col,FormGroup,Input,Row,Form, Button, Label} from 'reactstrap';
import SplitRegister from "./SplitRegister";

class Register extends Component{
    state={
        fullName:'',
        fullNameError:'',
        email:'',
        emailError:'',
        password:'',
        passwordError:'',
        phoneNumber:'',
        phoneNumberError:'',
        address:'',
        addressError:'',
        city:'',
        cityError:'',
        pincode:'',
        pincodeError:'',
        state:'',
        stateError:'',
        location:"",
        locationError:'',
        gender:{male:false,
            female:false,
        },
        genderError:'',
        checkbox:{
            morning:false,
            evening:false,
        },
        checkboxError:'',
        
    }

    radioBlur(){
        let errors = this.state;
        let male =  document.getElementById('male').checked;
        let female = document.getElementById('female').checked;
        if(!male && ! female){
            errors.genderError = "Select Gender.!";
        }
        else{
            errors.genderError= '';
        }
        this.setState({
            ...errors
        })
    }
    checkboxBlur= ()=>{
        let errors = this.state;
        let morning = document.getElementById('morning').checked;
        let evening = document.getElementById('evening').checked;
        if( !morning && !evening ){
            errors.checkboxError = "Pick the time slot";
        }
        else{
            errors.checkboxError='';
        }
        this.setState({
            ...errors
        })
    }

    handleBlur=(name,value)=>{
        let errors = this.state;
        if(name == 'fullName'){
            if(value == ''){
                errors.fullNameError="FullName is Required.!";
            }
            else{
                errors.fullNameError="";
            }
        }
        if(name == 'gender'){
            this.radioBlur();
        }
        if(name == 'checkbox'){
            this.checkboxBlur();
        }
        if(name == 'password'){
            if(value == ''){
                errors.passwordError="Password is Required.!";
            }
            else{
                errors.passwordError="";
            }    
        }
        if(name == 'phoneNumber'){
            if(value == ''){
                errors.phoneNumberError="PhoneNumber is Required.!";
            }
            else{
                errors.phoneNumberError="";
            }
        }
        if(name == 'address'){
            if(value == ''){
                errors.addressError="Address is Required.!";
            }
            else{
                errors.addressError="";
            }
        }
        if(name == 'state'){
            if(value == ''){
                errors.stateError="state is Required.!";
            }
            else{
                errors.stateError="";
            }  
        }
        if(name == 'city'){
            if(value == ''){
                errors.cityError="City is Required.!";
            }
            else{
                errors.cityError="";
            } 
        }
        if(name == 'pincode'){
            if(value == ''){
                errors.pincodeError="Pincode is Required.!";
            }
            else{
                errors.pincodeError="";
            }   
        }
        if(name == 'location'){
            if(value == ''){
                errors.locationError="Select location.!"
            }
            else{
                errors.locationError="";
            }
        }
        if(name == 'email'){
            if(!value){
                errors.emailError="Email is Required.!";
            }
            else{
                let model = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
                if(!model.test(value)){
                    errors.emailError="Invalid Email ID";
                }
                else{
                    errors.emailError='';
                }
            }
        }
        this.setState({
            ...errors
        })
    }
    
    render(){
        return(
            <Row>
                <Col>
                    <Card style={{marginTop:100,marginLeft:500,marginRight:300}}>
                        <CardHeader>
                            <div className='header' style={{marginLeft:100}}><b>REGISTRATION</b></div>
                        </CardHeader>
                        <CardBody>
                            <SplitRegister values={this.state} handleBlur={this.handleBlur}/>
                        </CardBody>
                    </Card>
                </Col>
            </Row>
            
        )
    }
}
export default Register