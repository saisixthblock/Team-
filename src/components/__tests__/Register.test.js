import React from "react";
import Register from "../Register";
import Enzyme from 'enzyme';
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Input } from "reactstrap";

Enzyme.configure({adapter:new Adapter() });

describe('Create a snapShot for the register component',()=>{
    it('Create the snapshot for the register component',()=>{
        let wrapper=shallow(<Register/>)
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});
describe('testing the register component',()=>{
    let wrapper=shallow(<Register/>)
    let instanceFun=wrapper.instance();
    let sampleData={
        fullName:'abcd',
        email:'abcd@gmail.com',
        password:'abcd',
        phoneNumber:'998877',
        address:'',
        city:'hyd',
        pincode:'502032',
        state:'',
        location:'',
        gender:"",
        checkbox:'',
    }

    beforeEach(()=>{
        jest.spyOn(instanceFun,'handleBlur');
        instanceFun.handleBlur('fullName','chandu');
        instanceFun.handleBlur('email','chandu@gmail.com')
    });
    it('check the fullname input field',()=>{
        expect(instanceFun.handleBlur).toHaveBeenCalledWith('fullName','chandu');
    });
    it('submit the user data',()=>{
        wrapper.setState({
            state:{...sampleData},
        });
    });
    it('check the IF condition',()=>{
        //console.log(instanceFun);
        //instanceFun.radioBlur();
    })
});