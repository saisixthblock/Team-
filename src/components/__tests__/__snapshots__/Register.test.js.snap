// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a snapShot for the register component Create the snapshot for the register component 1`] = `
<Row
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
      "xxl",
    ]
  }
>
  <Col
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Card
      style={
        Object {
          "marginLeft": 500,
          "marginRight": 300,
          "marginTop": 100,
        }
      }
      tag="div"
    >
      <CardHeader
        tag="div"
      >
        <div
          className="header"
          style={
            Object {
              "marginLeft": 100,
            }
          }
        >
          <b>
            REGISTRATION
          </b>
        </div>
      </CardHeader>
      <CardBody
        tag="div"
      >
        <SplitRegister
          handleBlur={[Function]}
          values={
            Object {
              "address": "",
              "addressError": "",
              "checkbox": Object {
                "evening": false,
                "morning": false,
              },
              "checkboxError": "",
              "city": "",
              "cityError": "",
              "email": "",
              "emailError": "",
              "fullName": "",
              "fullNameError": "",
              "gender": Object {
                "female": false,
                "male": false,
              },
              "genderError": "",
              "location": "",
              "locationError": "",
              "password": "",
              "passwordError": "",
              "phoneNumber": "",
              "phoneNumberError": "",
              "pincode": "",
              "pincodeError": "",
              "state": "",
              "stateError": "",
            }
          }
        />
      </CardBody>
    </Card>
  </Col>
</Row>
`;
